commands/user_test.go

```diff
@@ -0,0 +1,55 @@
+package commands
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mmctl/printer"
+
+	"github.com/golang/mock/gomock"
+	"github.com/spf13/cobra"
+)
+
+func (s *MmctlTestSuite) TestSearchUserCmd() {
+	s.Run("Search for an existing user", func() {
+		emailArg := "example@example.com"
+		mockUser := model.User{Username: "ExampleUser", Email: emailArg}
+
+		s.client.
+			EXPECT().
+			GetUserByEmail(gomock.Eq(emailArg), gomock.Eq("")).
+			Return(&mockUser, &model.Response{Error: nil}).
+			Times(1)
+
+		err := searchUserCmdF(s.client, &cobra.Command{}, []string{emailArg})
+		s.Require().Nil(err)
+		s.Require().Equal(&mockUser, printer.GetLines()[0])
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Search for a nonexistent user", func() {
+		printer.Clean()
```

Comment:
I'm a little torn here. It seems that none of the methods of `suite` would allow us to hook at the beginning/end of a `s.Run`, so there is no way to seamlessly reset the `printer` state between subtests.

A possible solution is this, if you are using subtests, you have to manually reset the `printer` state. Another solution would be that the `printer.GetLines` and `printer.GetErrorLines` methods would reset the lines as well, maybe renamed as `printer.GetLinesAndClean`.

Any strong opinion on this?

Reply:
After thinking about this for a while, we are going to stumble upon cases of subtests that don't check a type of lines, hence the next subtest will need to call `printer.Clean` anyways.